File: top_interview_questions/python/118_pascal_triangle.py
#!/usr/bin/env python
# encoding: utf-8
'''
@author: Fizz Cao 
@file: 118_pascal_triangle.py
@time: 12/10/2021 11:46 pm
'''
from typing import List


class Solution:
    def generate(self, numRows: int) -> List[List[int]]:
        ans = []

        for i in range(numRows):
            tool = [1] * (i + 1)
            for j in range(1, i):
                tool[j] = ans[i - 1][j] + ans[i - 1][j - 1]
            ans.append(tool)

        return ans

File: top_interview_questions/javascript/118_pascal_triangle.js
`
@author: Fizz Cao 
@file: 118_pascal_triangle.py
@time: 12/10/2021 11:46 pm
`
/**
 * @param {number} numRows
 * @return {number[][]}
 */
var generate = function(numRows) {
    let ans = [];

    for (let i = 0; i < numRows; i++) {
        let tool = [];
        for (let j = 0; j < i+1; j++)
            tool.push(1);
        for (let k = 1; k < i ; k++)
            tool[k] = ans[i-1][k] + ans[i-1][k-1];
        ans.push(tool);
    }

    return ans;
};